File: samples-node/datalayer.provider.alldata/src/datatype.ts
/*
 * SPDX-FileCopyrightText: Bosch Rexroth AG
 *
 * SPDX-License-Identifier: MIT
 */



/**
 * The interface IDataType.
 */
interface IDataType {
    /**
     * Gets the name.
     */
    getName(): string

    /**
     * Gets the variant type.
     */
    getVariantType(): string

    /**
     * Gets the target address.
     */
    getTargetAddress(): string

    /**
     * Gets the data type as string.
     */
    toString(): string
}

/**
 * The class DataType.
 */
class DataType implements IDataType {

    static rootType = 'types/sdk/node/provider/all-data';

    /** 
     * Creates a new instance of the DataType class
     */
    constructor(
        private name: string,
        private variantType: string,
        private targetAddress: string
    ) { }

    /**
     * Factory to create a new instance of the DataType class.
     * @param name 
     * @param variantType 
     * @param targetAddress 
     * @returns 
     */
    static create(name: string, variantType: string, targetAddress: string): DataType {
        return new DataType(name, variantType, targetAddress);
    }

    /**
    * Gets the name.
    * @returns 
    */
    getName(): string {
        return this.name;
    }

    /**
    * Gets the variant type.
    * @returns 
    */
    getVariantType(): string {
        return this.variantType;
    }

    /**
    * Gets target address.
    * @returns 
    */
    getTargetAddress(): string {
        return this.targetAddress;
    }

    /**
    * Stringifies the DataType class
    * @returns 
    */
    toString(): string {
        return this.name;
    }
}

export = DataType

File: samples-net/datalayer.provider.alldata/DataType.cs
/*
 * SPDX-FileCopyrightText: Bosch Rexroth AG
 *
 * SPDX-License-Identifier: MIT
 */
namespace Samples.Datalayer.Provider.Alldata
{
    /// <summary>
    /// Defines the <see cref="DataType" />.
    /// </summary>
    internal class DataType
    {
        /// <summary>
        /// Gets the Name.
        /// </summary>
        public string Name { get; }

        /// <summary>
        /// Gets the Address.
        /// </summary>
        public string Address { get; }

        /// <summary>
        /// Initializes a new instance of the <see cref="DataType"/> class.
        /// </summary>
        /// <param name="name">The name<see cref="string"/>.</param>
        /// <param name="address">The address<see cref="string"/>.</param>
        public DataType(string name, string address)
        {
            Name = name;
            Address = address;
        }
    }
}
